src/devices/src/virtio/net/device.rs

```diff
@@ -254,20 +252,23 @@ impl Net {
             DeviceError::FailedSignalingIrq(e)
         })?;
 
-        self.rx_deferred_irqs = false;
         Ok(())
     }
 
     fn signal_rx_used_queue(&mut self) -> result::Result<(), DeviceError> {
-        if !self.rx_deferred_irqs {
-            return Ok(());
+        let mem = self.device_state.mem().unwrap();
+        if self.queues[RX_INDEX].prepare_kick(mem) {
+            return self.signal_used_queue();
         }
 
+        Ok(())
+    }
+
+    fn signal_tx_used_queue(&mut self) -> result::Result<(), DeviceError> {
         let mem = self.device_state.mem().unwrap();
-        if self.queues[RX_INDEX].prepare_kick(mem) {
+        if self.queues[TX_INDEX].prepare_kick(mem) {
             return self.signal_used_queue();
         }
```

Comment:
Can we have a single method that receives the queue as a parameter ? Probably as an enum.

Reply:
I added a commit that does that.